core/API/Request.php

```diff
@@ -412,6 +412,9 @@ public static function getRequestParametersGET()
             return array();
         }
         $GET = UrlHelper::getArrayFromQueryString($_SERVER['QUERY_STRING']);
+        foreach ($GET as $name => $value) {
+            $GET[$name] = urldecode($value);
+        }
```

Comment:
We should make this method private, to make sure nobody else can call it but it is an API method. So we will break API here, maybe it's not used though. Alternatively we could leave this method as it is, and add a new private method like `getUrlDecodedRequestParametersGET`. As long as it is only used for segments it might not introduce a security regression.